Projects/Contracts/Sources/View/CoInsured/CoInusuredInput.swift

```diff
@@ -0,0 +1,273 @@
+        contractId: String
+    ) {
+        self.isDeletion = isDeletion
+        self.fullName = fullName ?? ""
+        self.firstName = ""
+        self.lastName = ""
+        self.SSN = SSN ?? ""
+        self.contractId = contractId
+    }
+
+    var body: some View {
+        hForm {
+            VStack(spacing: 4) {
+                if isDeletion {
+                    deleteCoInsuredFields
+                } else {
+                    addCoInsuredFields
+                }
+                hSection {
+                    LoadingButtonWithContent(ContractStore.self, .fetchNameFromSSN) {
+                        if !(buttonIsDisabled || nameFetchedFromSSN || noSSN) {
+                            /* TODO: FETCH SSN MUTATION*/
+                            DispatchQueue.main.asyncAfter(deadline: .now() + 3.0) {
+                                //on success - show name field
+                                // else show error
+                                nameFetchedFromSSN = true
+                                firstName = "Hedvig"
+                                lastName = "AB"
+                            }
+                        } else if isDeletion {
+                            store.coInsuredViewModel.removeCoInsured(name: fullName, personalNumber: SSN)
+                            store.send(.coInsuredNavigationAction(action: .deletionSuccess))
+                        } else if nameFetchedFromSSN || noSSN {
+                            store.coInsuredViewModel.addCoInsured(name: firstName + " " + lastName, personalNumber: SSN)
+                            store.send(.coInsuredNavigationAction(action: .addSuccess))
+                        }
+                    } content: {
+                        hText(buttonDisplayText)
+                            .transition(.opacity.animation(.easeOut))
+                    }
+                }
+                .padding(.top, 12)
+                .disabled(buttonIsDisabled && !isDeletion)
+
+                hButton.LargeButton(type: .ghost) {
+                    store.send(.coInsuredNavigationAction(action: .dismissEdit))
+                } content: {
+                    hText(L10n.generalCancelButton)
+                }
+                .padding(.horizontal, 16)
+                .padding(.top, 4)
+            }
+        }
+    }
+
+    var buttonDisplayText: String {
+        if isDeletion {
+            return L10n.removeConfirmationButton
+        } else if nameFetchedFromSSN {
+            return L10n.contractAddCoinsured
+        } else if Masking(type: .personalNumber).isValid(text: SSN) {
+            return L10n.contractSsnFetchInfo
+        } else if buttonIsDisabled || noSSN {
+            return L10n.generalSaveButton
+        }
+        return ""
+    }
+
+    @ViewBuilder
+    var addCoInsuredFields: some View {
+        Group {
+            ZStack {
+                if noSSN {
+                    hSection {
+                        hFloatingTextField(
```

Comment:
why we dont use date picker here?

Reply:
Since we pick date of a birth. :/ make more sense

Reply:
Let's go with the design now and we can discuss this with William :)

Reply:
oki